lib/timeout.rb

```diff
@@ -25,27 +25,31 @@
 module Timeout
   VERSION = "0.3.2"
 
+  # Internal error raised to when a timeout is triggered.
+  class ExitException < Exception
+    # Return the receiver if it the exception is triggered in
+    # the same thread.
```

Comment:
Do we need this or could we just assume the ExitException we `rescue` is the one we raised? Maybe it's necessary to handle the case of nested timeouts?

Reply:
The docs say:
```
=== Implementation from Exception
------------------------------------------------------------------------
  exc.exception([string])  ->  an_exception or exc

------------------------------------------------------------------------

With no argument, or if the argument is the same as the receiver, return
the receiver. Otherwise, create a new exception object of the same class
as the receiver, but with a message equal to string.to_str.
```
so I think we don't need to do anything special here (it should already return the receiver).

Reply:
Agreed.

Reply:
Turns out the `#exception` define is needed, because the `interrupt` method does `@thread.raise @exception_class, @message`.  Without the `@message`, it wouldn't be needed.  However, the `@message` is needed in the cases where you are providing a `klass` argument to `timeout`.